pdns/rcpgenerator.cc

```diff
@@ -362,13 +362,21 @@ void RecordTextReader::xfrSvcParamKeyVals(set<SvcParam>& val) // NOLINT(readabil
       throw RecordTextException(e.what());
     }
 
-    if (key != SvcParam::no_default_alpn) {
-      if (d_pos == d_end || d_string.at(d_pos) != '=') {
-        throw RecordTextException("expected '=' after " + k);
+    if (key == SvcParam::no_default_alpn) {
+      if (d_pos != d_end) {
```

Comment:
This rejects `1 foo.powerdns.org. key2 ipv4hint=1.2.3.4` while accepting `1 foo.powerdns.org. ipv4hint=1.2.3.4 key2`